Binary files csvfile_formatter_org/.DS_Store and csvfile_formatter_changed/.DS_Store differ
diff -ruN csvfile_formatter_org/css/csvfile-formatter.css csvfile_formatter_changed/css/csvfile-formatter.css
--- csvfile_formatter_org/css/csvfile-formatter.css	1969-12-31 19:00:00
+++ csvfile_formatter_changed/css/csvfile-formatter.css	2024-08-23 10:33:16
@@ -0,0 +1,43 @@
+table.dataTable thead .sorting {
+	vertical-align: middle !important;
+}
+
+#accordion .card .card-header [aria-expanded=true], #additional-accordion .card .card-header [aria-expanded=true] {
+    border: 3px solid #337ab7;
+    background-color: #002a5c;
+    color: #fff;
+}
+
+#accordion accordion.card .card-header .btn, #additional-accordion .card .card-header .btn {
+    text-align: left;
+    background-color: #dae1e9;
+    color: #333;
+    font-size: 18px;
+    font-size: 1.8rem;
+    border-radius: 0;
+    -webkit-border-radius: 0;
+    -moz-border-radius: 0;
+    padding: 1rem;
+    text-decoration: none;
+    width: 100%;
+    border: 3px solid rgba(0, 0, 0, 0);
+    white-space: normal;
+    cursor: pointer;
+    display: flex;
+    align-items: center;
+}
+
+.truncate {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    max-width: 200px;
+    display: inline-block;
+}
+.full-text {
+    display: none;
+}
+
+a.toggle-text {
+    color:#fff !important;
+}
\ No newline at end of file
diff -ruN csvfile_formatter_org/csvfile_formatter.libraries.yml csvfile_formatter_changed/csvfile_formatter.libraries.yml
--- csvfile_formatter_org/csvfile_formatter.libraries.yml	2023-06-01 10:50:37
+++ csvfile_formatter_changed/csvfile_formatter.libraries.yml	2024-08-23 10:32:08
@@ -3,7 +3,7 @@
   js:
     js/csvfile-formatter.js: {}
   css:
-    component:
+    theme:
       css/csvfile-formatter.css: {}
   dependencies:
     - core/jquery
diff -ruN csvfile_formatter_org/js/csvfile-formatter.js csvfile_formatter_changed/js/csvfile-formatter.js
--- csvfile_formatter_org/js/csvfile-formatter.js	2023-06-01 10:50:37
+++ csvfile_formatter_changed/js/csvfile-formatter.js	2024-08-23 10:32:08
@@ -3,37 +3,77 @@
  * CSV File Formatter behaviors.
  */
 
- (function ($, Drupal) {
+(function ($, Drupal) {
 
-  'use strict';
-
-  /**
-   * Adds DataTable to tables with the class '.add-externaljs-csvfiletable'.
-   */
-  Drupal.behaviors.csvfileFormatter = {
-    attach: function (context, settings) {
-      var dataTableSettings = drupalSettings.csvFormatter.dataTable;
-
-      var elements = once('csvfiletable-once', jQuery('.add-externaljs-csvfiletable'), context);
-      elements.forEach(element => {
-        var dataTableSetting = Object.assign({}, dataTableSettings[0]);
-        var numColumns = jQuery(element).find('th').length;
-
-        var scrollXMin = dataTableSetting['scrollX'];
-        if (scrollXMin >= 0 && numColumns >= scrollXMin) {
-          dataTableSetting['scrollX'] = true;
-        }
-        else {
-          dataTableSetting['scrollX'] = false;
-        }
-
-        if (dataTableSetting['scrollY'] < 0) {
-          delete dataTableSetting.scrollY;
-        }
-
-        jQuery(element).DataTable(dataTableSetting);
-      })
-    }
-  };
-
-} (jQuery, Drupal));
+    'use strict';
+  
+    /**
+     * Adds DataTable to tables with the class '.add-externaljs-csvfiletable'.
+     */
+    Drupal.behaviors.csvfileFormatter = {
+      attach: function (context, settings) {
+          var dataTableSettings = drupalSettings.csvFormatter.dataTable;
+  
+          var elements = once('csvfiletable-once', jQuery('.add-externaljs-csvfiletable'), context);
+          elements.forEach(element => {
+              var dataTableSetting = Object.assign({}, dataTableSettings[0]);
+              var numColumns = jQuery(element).find('th').length;
+  
+              var scrollXMin = dataTableSetting['scrollX'];
+              if (scrollXMin >= 0 && numColumns >= scrollXMin) {
+                  dataTableSetting['scrollX'] = true;
+              }
+              else {
+                  dataTableSetting['scrollX'] = false;
+              }
+  
+              if (dataTableSetting['scrollY'] < 0) {
+                  delete dataTableSetting.scrollY;
+              }
+  
+              // Add columnDefs to dataTableSetting
+              dataTableSetting['columnDefs'] = [{
+                  orderable: false,
+                  targets: '_all', // replace with your target column number
+                  render: function (data, type, row) {
+                      if (type === 'display' && data.length > 50) {
+                          return '<span class="truncate">' +data.substr(0, 47) + '</span><span class="full-text">'+data+'</span> <a href="#" class="toggle-text">more....</a>';
+                      } else {
+                          return data;
+                      }
+                  }
+              }];
+              // Add header rendering function
+              dataTableSetting['headerCallback'] = function (thead, data, start, end, display) {
+                  $(thead).find('th').each(function () {
+                      var $this = $(this);
+                      var text = $this.text();
+                      if (text.length > 50) {
+                          $this.html('<span class="truncate">' + text.substr(0, 47) + '</span><span class="full-text">' + text + '</span> <a href="#" class="toggle-text">more...</a>');
+                      }
+                  });
+              };
+              var table = jQuery(element).DataTable(dataTableSetting);
+              $('.toggle-text').on('click', function(e) {
+                  e.preventDefault();
+                  var $this = $(this);
+                  var $truncate = $this.siblings('.truncate');
+                  var $fullText = $this.siblings('.full-text');
+  
+                  if ($truncate.is(':visible')) {
+                      $truncate.hide();
+                      $fullText.show();
+                      $this.text('Show less');
+                  } else {
+                      $truncate.show();
+                      $fullText.hide();
+                      $this.text('more...');
+                  }
+              });
+          });
+  
+      }
+    };
+  
+  } (jQuery, Drupal));
+  
\ No newline at end of file
